x-pack/legacy/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -235,6 +237,41 @@ export function XYChart({
         theme={chartTheme}
         rotation={shouldRotate ? 90 : 0}
         xDomain={xDomain}
+        onBrushEnd={(min: number, max: number) => {
+          // in the future we want to make it also for histogram
+          if (!xAxisColumn || xAxisColumn.meta?.type !== 'date_histogram') {
+            return;
+          }
+
+          const table = data.tables[layers[0].layerId];
+          const xAxisFieldName: string | undefined = xAxisColumn?.meta?.aggConfigParams?.field;
```

Comment:
By accessing the `aggConfigParams` property of the table, you are breaking one of the rules that we decided to stick with in Lens: visualizations aren't allowed to write code that is specific to a single datasource. The way we usually would write this code as a result is that instead of checking for `aggConfigParams` or the `date_histogram` function, you would check for `xScaleType === 'time'` or `dataType === 'date'`

Reply:
@wylieconlon but this is the information we need to pass timeFieldName, so the filter will be applied to time picker and not as a regular filter. There's no other way to get it from config. We do the same thing onClick. Do you think we should extend configuration and add xAxisFieldName?

Reply:
as discussed in the meeting this should be fine as we will be moving field information off the aggConfigParams directly to meta and is something that can be provided by most datasources.